### PR TITLE
isDeletableFile Implementation + Tests

### PR DESCRIPTION
**Pull request by rauhul**
Tried to best match the suggested implementation from the [Swift Forum](https://forums.swift.org/t/swiftfoundation-filemanager-isdeletablefile-implementation/13465/7).

I wasn't exactly sure how to write tests for these methods. My tests attempt to check that the result of `isReadableFile`, `isWritableFile`, `isExecutableFile` match the value one would receive by directly checking the file's permission bits.

### CODE REVIEW

File: TestFoundation/TestFileManager.swift [hunk 1]
```diff
@@ -202,6 +206,62 @@ class TestFileManager : XCTestCase {
...
+            XCTAssertTrue(fileIsReadableFile == fmIsReadableFile)
+        } catch let e {
+            XCTFail("\(e)")
+        }
+    }
+
+    func test_isDeletableFile() {
+        // TODO: Implement test
+        // how to test?
+    }
```
Comment (spevans) [inline TestFoundation/TestFileManager.swift]: You could create a file and check its is deletable. Also check a file that should fail eg `/dev/null` and also check `/` to validate the parent checking code

Comment (rauhul) [reply TestFoundation/TestFileManager.swift]: Just to confirm, we would expect the created file to be deletable, correct?

Comment (spevans) [reply TestFoundation/TestFileManager.swift]: Yes, thats correct. You should be able to create a file that fails by putting it into a subdirectory with no permissions:

eg 
```
mkdir dir1
touch dir1/file1
chmod 000 dir1

isDeletableFile("dir1/file1") -> false

chmod 755 dir1

isDeletableFile("dir1/file1") -> true
```

**Comment by rauhul** [comment]
@spevans does the `test_isDeletableFile` implementation look good?

**Comment by spevans** [comment]
@swift-ci test

File: TestFoundation/TestFileManager.swift [hunk 2]
```diff
@@ -225,41 +225,61 @@ class TestFileManager : XCTestCase {
...
         do {
             let attrs = try fm.attributesOfItem(atPath: path)
             let permissions = attrs[FileAttributeKey.posixPermissions] as! UInt16
-            let fileIsReadableFile = (permissions & S_IXUSR == S_IXUSR)
-            let fmIsReadableFile = fm.isReadableFile(atPath: path)
-            XCTAssertTrue(fileIsReadableFile == fmIsReadableFile)
+            let fileIsExecutableFile = (permissions & S_IXUSR == S_IXUSR)
+            let fmIsExecutableFile = fm.isExecutableFile(atPath: path)
+            XCTAssertTrue(fileIsExecutableFile == fmIsExecutableFile)
         } catch let e {
```
Comment (spevans) [inline TestFoundation/TestFileManager.swift]: I would modify this test to set the executable bit on the file and then simply test it with:
`XCTAssertTrue(fm.isExecutableFile(atPath: path))` (both before and after setting the bit)

Testing equality this way could mask an issue if the `let fileIsExecutableFile = (permissions & S_IXUSR == S_IXUSR)` was wrong.

Comment (rauhul) [reply TestFoundation/TestFileManager.swift]: Makes sense, will do

**Comment by spevans** [comment]
@swift-ci test

**Comment by rauhul** [comment]
@parkera @spevans can this be merged?

**Comment by alblue** [comment]
Seems reasonable to me. WDYT @millenomi @phausler ?

**Comment by parkera** [comment]
@millenomi you want to do the final approval here?

File: Foundation/FileManager.swift [hunk 3]
```diff
@@ -744,9 +744,41 @@ open class FileManager : NSObject {
...
+    /**
+     - parameters:
+        - path: The path to the file we are trying to determine is deletable.
+
+      - returns: `true` if the file is deletable, `false` otherwise.
+     */
     open func isDeletableFile(atPath path: String) -> Bool {
-        NSUnimplemented()
+        // Get the parent directory of supplied path
+        let parent = path._nsObject.deletingLastPathComponent
```
Comment (millenomi) [inline Foundation/FileManager.swift]: This is pretty close to the Darwin implementation! That implementation also special-cases the empty string to mean the current directory, but it's not behavior that's documented, so I don't think we need to port it.

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Pull request merged**